drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c

```diff
@@ -1062,12 +1062,15 @@ static int mmal_setup_components(struct bm2835_mmal_dev *dev,
 	case COMP_CAMERA:
 		/* Make a further decision on port based on resolution */
 		if (f->fmt.pix.width <= max_video_width &&
-		    f->fmt.pix.height <= max_video_height)
-			camera_port = port =
+		    f->fmt.pix.height <= max_video_height) {
+			camera_port =
 			    &dev->component[COMP_CAMERA]->output[CAM_PORT_VIDEO];
-		else
-			camera_port = port =
+			port = camera_port;
```

Comment:
Move the duplicated assignment to port out outside the conditional and avoid the extra braces.